app/controllers/admin/html_variants_controller.rb

```diff
@@ -0,0 +1,84 @@
+module Admin
+  class HtmlVariantsController < Admin::ApplicationController
+    layout "admin"
+
+    def index
+      relation = if params[:state] == "mine"
+                   current_user.html_variants.order(created_at: :desc)
+                 elsif params[:state] == "admin"
+                   HtmlVariant.where(published: true, approved: false).order(created_at: :desc)
+                 elsif params[:state].present?
+                   HtmlVariant.where(published: true, approved: true, group: params[:state]).order(success_rate: :desc)
+                 else
+                   HtmlVariant.where(published: true, approved: true).order(success_rate: :desc)
+                 end
+
+      @html_variants = relation.includes(:user).page(params[:page]).per(30)
+    end
+
+    def new
+      @html_variant = HtmlVariant.new
+      return unless params[:fork_id]
+
+      @fork = HtmlVariant.find(params[:fork_id])
+      @html_variant.name = @fork.name + " FORK-#{rand(10_000)}"
```

Comment:
**nitpick (non-blocking):** Since you're already interpolating on the left, let's save one more intermediate string object:

```suggestion
      @html_variant.name = "#{@fork.name} FORK-#{rand(10_000)}"
```